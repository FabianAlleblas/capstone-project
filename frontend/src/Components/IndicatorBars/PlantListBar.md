PlantListBar: normal

```js
<PlantListBar
  daysLeft={3}
  weeksLeft={2}
  waterInterval={10}
  fertilizerInterval={4}
/>
```

PlantListBar: low

```js
<PlantListBar
  daysLeft={0}
  weeksLeft={1}
  waterInterval={10}
  fertilizerInterval={4}
/>
```
