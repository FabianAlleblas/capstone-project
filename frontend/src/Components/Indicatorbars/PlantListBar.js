import PropTypes from 'prop-types'
import styled from 'styled-components'
import { ReactComponent as FertilizerIcon } from '../../assets/icons/fertilizericon.svg'
import { ReactComponent as WaterIcon } from '../../assets/icons/watericon.svg'

export default function PlantListBar({ daysLeft, weeksLeft }) {
  PlantListBar.propTypes = {
    daysLeft: PropTypes.number.isRequired,
    weeksLeft: PropTypes.number.isRequired,
  }

  return (
    <Container>
      <WaterIcon />
      <Bar>
        <WaterIndicator daysLeft={daysLeft} />
      </Bar>
      <FertilizerIcon />
      <Bar>
        <FertilizerIndicator weeksLeft={weeksLeft} />
      </Bar>
    </Container>
  )
}

const Container = styled.div`
  display: grid;
  gap: 6px 6px;
  grid-template-columns: 15px auto;
  place-items: center;
  width: 100%;
  padding: 10px 0 0;
`

const Bar = styled.div`
  background-color: var(--bar-bg-color);
  border-radius: 1px;
  height: 2px;
  position: relative;
  width: 100%;
`

const WaterIndicator = styled.div`
  background-color: ${(props) =>
    props.daysLeft < 2 ? 'var(--warning-color)' : 'var(--bar-water-color)'};
  border-radius: 3px;
  height: 6px;
  min-width: 5px;
  position: absolute;
  top: -2px;
  width: ${(props) => (100 / 10) * props.daysLeft}%;
`

const FertilizerIndicator = styled.div`
  background-color: ${(props) =>
    props.weeksLeft < 2
      ? 'var(--warning-color)'
      : 'var(--bar-fertilizer-color)'};
  border-radius: 3px;
  height: 6px;
  min-width: 5px;
  position: absolute;
  top: -2px;
  width: ${(props) => (100 / 4) * props.weeksLeft}%;
`
