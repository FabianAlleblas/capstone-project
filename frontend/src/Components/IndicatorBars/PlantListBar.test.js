import { render } from '@testing-library/react'
import 'jest-styled-components'
import PlantListBar from './PlantListBar'

describe('PlantDetailBar', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PlantListBar
        daysLeft={10}
        weeksLeft={4}
        waterInterval={10}
        fertilizerInterval={4}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('renders with the full length', () => {
    const { getByTestId } = render(
      <PlantListBar
        daysLeft={10}
        weeksLeft={4}
        waterInterval={10}
        fertilizerInterval={4}
      />
    )

    const WaterBar = getByTestId('water-indicator')
    const FertilizerBar = getByTestId('fertilizer-indicator')

    expect(WaterBar).toHaveStyle('width: 100%;')
    expect(FertilizerBar).toHaveStyle('width: 100%')
  })
  it('renders with the min length', () => {
    const { getByTestId } = render(
      <PlantListBar
        daysLeft={0}
        weeksLeft={0}
        waterInterval={10}
        fertilizerInterval={4}
      />
    )

    const WaterBar = getByTestId('water-indicator')
    const FertilizerBar = getByTestId('fertilizer-indicator')

    expect(WaterBar).toHaveStyle('width: 0%;')
    expect(FertilizerBar).toHaveStyle('width: 0%')
  })
})
