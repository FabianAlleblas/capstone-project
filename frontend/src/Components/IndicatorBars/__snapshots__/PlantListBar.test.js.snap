// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantDetailBar renders correctly 1`] = `
.c0 {
  display: grid;
  gap: 6px 6px;
  grid-template-columns: 15px auto;
  padding: 10px 0 0;
  place-items: center;
  width: 100%;
}

.c1 {
  background-color: var(--bar-bg-color);
  border-radius: 1px;
  height: 2px;
  position: relative;
  width: 100%;
}

.c2 {
  border-radius: 3px;
  height: 6px;
  min-width: 5px;
  position: absolute;
  top: -2px;
}

.c3 {
  background-color: var(--bar-water-color);
  width: 100%;
}

.c4 {
  background-color: var(--bar-fertilizer-color);
  width: 100%;
}

<div
  class="c0"
>
  <svg>
    watericon.svg
  </svg>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
      data-testid="water-indicator"
    />
  </div>
  <svg>
    fertilizericon.svg
  </svg>
  <div
    class="c1"
  >
    <div
      class="c2 c4"
      data-testid="fertilizer-indicator"
    />
  </div>
</div>
`;
